binding/python/test_rhino.py

```diff
@@ -39,77 +46,133 @@ def _context_path():
+                                        contexts_root, 'beaglebone', f'{context}_beaglebone.rhn')
                 else:
                     raise NotImplementedError("Unsupported CPU: '%s'." % cpu_part)
         elif system == 'Windows':
-            return os.path.join(os.path.dirname(__file__), '../../resources/contexts/windows/coffee_maker_windows.rhn')
+            return os.path.join(os.path.dirname(__file__), contexts_root, 'windows', f'{context}_windows.rhn')
         else:
             raise ValueError("Unsupported system '%s'." % system)
 
-    rhino = None
+    @classmethod
+    def __pv_model_path_by_language(cls, relative, language):
+        model_path_subdir = cls.__append_language('lib/common/rhino_params', language)
+        model_path_subdir = f'{model_path_subdir}.pv'
+        return os.path.join(os.path.dirname(__file__), relative, model_path_subdir)
+
+    rhinos = None
 
     @classmethod
     def setUpClass(cls):
-        cls.rhino = Rhino(
-            access_key=sys.argv[1],
-            library_path=pv_library_path('../..'),
-            model_path=pv_model_path('../..'),
-            context_path=cls._context_path())
+        _language_to_contexts = {
+            'en': ['coffee_maker'],
+            'es': ['luz'],
+            'de': ['beleuchtung']
+        }
+
+        cls.rhinos = dict()
+        for language in _language_to_contexts:
+            cls.rhinos[language] = dict()
+            for context in _language_to_contexts[language]:
+                cls.rhinos[language][context] = Rhino(
+                    access_key=sys.argv[1],
+                    library_path=pv_library_path('../..'),
+                    model_path=cls.__pv_model_path_by_language('../..', language),
+                    context_path=cls.__context_path(context, language)
+                )
 
     @classmethod
     def tearDownClass(cls):
-        if cls.rhino is not None:
-            cls.rhino.delete()
+        if cls.rhinos is not None:
+            for language in cls.rhinos:
+                for context in cls.rhinos[language]:
+                    cls.rhinos[language][context].delete()
+
+    def run_rhino(self, language, context, is_whithin_context, intent=None, slots=None,
+                  audio_file_name=None):
+        rhino = self.rhinos[language][context]
+
+        if audio_file_name is None:
+            if is_whithin_context:
+                prefix = 'test_within_context'
+            else:
+                prefix = 'test_out_of_context'
+            audio_file_name = f'{self.__append_language(prefix, language)}.wav'
 
-    def test_within_context(self):
         audio, sample_rate = \
             soundfile.read(
-                os.path.join(os.path.dirname(__file__), '../../resources/audio_samples/test_within_context.wav'),
+                os.path.join(os.path.dirname(__file__), '../../resources/audio_samples/', audio_file_name),
```

Comment:
I am ok with changes, just that we got rid of soundfile in the demos (issues on mac arm). @laves do you think we should do the same here?

Reply:
probably, but we can add that to tracking as a new task